src/routes/blogs/olive-quant-ft/+page.svx

```diff
@@ -0,0 +1,196 @@
+---
+title: 'Is it better to quantize before or after finetuning?'
+date: '18th November, 2024'
+description: 'Learn how to use the shared cache feature in Olive to enhance team collaboration when optimizing AI models'
+keywords: 'GenAI , LLM, ONNXRuntime, ORT, Phi, DirectML, Windows, phi3, phi-3, llama-3.2, ONNX, SLM, edge, gpu'
+authors:
+  [
+    'Jambay Kinley',
+    'Sam Kemp'
+    ]
+authorsLink:
+  [
+    'https://www.linkedin.com/in/jambayk/',
+    'https://www.linkedin.com/in/samuel-kemp-a9253724/'
+   ]
+image: ''
+imageSquare: ''
+url: 'https://onnxruntime.ai/blogs/olive-quant-ft'
+---
+
+
+## 👋 Introduction
+
+Quantization in machine learning is a technique used to reduce the precision of the numbers used in computations, which helps in making models more efficient. Instead of using high-precision floating point numbers (like 32-bit or 16-bit), quantization converts these numbers to lower-precision formats, such as 8-bit integers. The primary benefits of quantization are a smaller model size and faster computations, which are particularly useful for deploying models on devices with limited resources, like mobile phones or embedded systems. However, this reduction in precision can sometimes lead to a slight decrease in the model's accuracy.
+
+Fine-tuning an AI model using the LoRA (Low-Rank Adaptation) method is an efficient way to adapt large language models to specific tasks or domains. Instead of retraining all the model parameters, LoRA modifies the fine-tuning process by freezing the original model weights and applying changes to a separate set of weights, which are then added to the original parameters. This approach transforms the model parameters into a lower-rank dimension, reducing the number of parameters that need training, thus speeding up the process and lowering costs.
+
+When fine-tuning and quantizing a model, it is important to establish the correct sequence:
+
+- Is it better to quantize *before* fine-tuning or after?
+
+In theory, quantizing before fine-tuning should yield a higher quality model because it provides an opportunity for the fine-tuning process to recover some of the loss from the quantization process. In this blog post we demonstrate how Olive - a state-of-the-art model optimization toolkit for the ONNX runtime - can help you answer when to quantize and which quantization algorithm to use for a given model architecture and scenario.
```

Comment:
recovering loss is one reason but the main reason is that since the base model is already in the final expected precision, there is no further loss in quality due to quantization. The adapters are trained on the quantized model and used with the quantized model, rather than trained on float model and used with a subsequently quantized model.